drivers/backup/backup.go

```diff
@@ -0,0 +1,104 @@
+package backup
+
+import (
+	"fmt"
+
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/torpedo/pkg/errors"
+)
+
+// Image Generic struct
+type Image struct {
+	Type    string
+	Version string
+}
+
+// Driver for backup
+type Driver interface {
+	// Init initializes the backup driver under a given scheduler
+	Init(schedulerDriverName string, nodeDriverName string, volumeDriverName string, token string) error
+
+	// String returns the name of this driver
+	String() string
+
+	// CreateOrganization creates Organization
+	CreateOrganization(orgName string) error
+
+	// GetOrganization enumerates organizations
+	GetOrganization() error
+
+	// CreateCloudCredential creates cloud credential objects
+	CreateCloudCredential(orgID string, name string) error
+
+	// InspectCloudCredential describes the cloud credential
+	InspectCloudCredential(orgID string, name string) (*api.CloudCredentialInspectResponse, error)
+
+	// EnumerateCloudCredential lists the cloud credentials for given Org
+	EnumerateCloudCredential(orgID string) (*api.CloudCredentialEnumerateResponse, error)
+
+	// DeletrCloudCredential deletes a cloud credential object
+	DeleteCloudCredential(orgID string, name string) error
+
+	// CreateCluster creates a cluste object
+	CreateCluster(name string, orgID string, labels map[string]string, cloudCredential string, pxToken string, config string) (*api.ClusterCreateResponse, error)
+
+	// EnumerateCluster enumerates the cluster objects
+	EnumerateCluster(OrgID string) (*api.ClusterEnumerateResponse, error)
+
+	// InsepctCluster describes a cluster
+	InspectCluster(OrgID string, name string) (*api.ClusterInspectResponse, error)
+
+	// DeleteCluster deletes a cluster object
+	DeleteCluster(orgID string, name string) (*api.ClusterDeleteResponse, error)
+
+	// CreateBackupLocation creates backup location object
+	CreateBackupLocation(orgID string, name string, cloudCredential string, path string, encKey string,
```

Comment:
For all the APIs, take in the Request structure instead of all the params individually, that way if new fields are added we don't need to keep updating the interfaces